### ISSUE TITLE
Include in Beaker 0.9?

### ISSUE THREAD

**Issue by pfrazee** [brechtcs/dterm#12]
Hey, just wanted to check in on dterm's status and see what you would think of possibly bundling dterm in beaker 0.9? There's a browser sidebar that I think this could be really cool for.

**Comment by brechtcs** [comment]
Yeah, I saw [your demo](https://twitter.com/pfrazee/status/1160732568616148992) on Twitter. It does look quite handy to have it in the sidebar like that, so I'm definitely interested.

As for the status of this project, I've been fiddling with the secure execution environment on my local machine (as discussed in https://github.com/dfurl/dterm/issues/8). I basically got the messaging in and out of the `iframe` sandbox working, including a shim of `DatArchive.readFile` and the `cat` command. I still have to finish all other `DatArchive` method shims to complete it, and then clean up the code a bit. Would this still be necessary though if we choose to bundle dterm as a sidebar app? Or do these have their own security sandbox already?

**Comment by pfrazee** [comment]
Yeah sorry about that, I left this comment and then thought "heck I can just pull out my old webterm code." I ended up making it *very* simple and featureless, and I think that might be the right approach to launch: no composition, a solid set of builtins, and a way to shell out to scripts.

I don't have a secure execution solution yet so that's the dterm work I'd be most interested in. I'd be happy to credit you in the code and in the webterm startup message.

I'm thinking we just fallback from the builtin commands by looking for possible scripts to run. The .js on the script-path would be optional and we'd resolve against a user directory of saved scripts. So...

```
browserify ./main.js                -- saved script
./scripts/browserify ./main.js      -- relative script
dat://foo.com/browserify ./main.js  -- fully-qualified script
```

**Comment by brechtcs** [comment]
With some small hacks I've managed to run the terminal view of [beaker-sidebar-app](https://github.com/beakerbrowser/beaker-sidebar-app) in Beaker 0.8.8. So if you want I can create a WIP pull request over there with the iframe sandbox code to get things started. I've got a holiday coming up, so should be able to spend some time on it this week.

**Comment by pfrazee** [comment]
Okay awesome, let me know how it goes
